src/socketServer.ts

```diff
@@ -0,0 +1,71 @@
+import dotenv from 'dotenv';
+import cors from 'cors';
+import express, { Express } from 'express';
+import { createServer, Server as HTTPServer } from "http";
+import { ClientToServerEvents, ServerToClientEvents } from './models/events.model';
+import { Server as SocketServer } from 'socket.io';
+
+/**
+ * Keeps track of application lifecycle and maintains a testable server context.
+ *
+ * Not all fields or methods should be exposed as the application context. If a field is not marked as `private`, document the reason.
+ */
+export class socketServer {
+    private readonly serverContext: HTTPServer;
+
+    private readonly port: string;
+
+    constructor() {
+        // Load environment variables
+        dotenv.config({ path: `${__dirname}/../config/.env` });
+        this.port = process.env.SOCKET_SERVER_PORT || "8900";
+
+        const expressApp: Express = express();
+        expressApp.use(cors());
+
+        this.serverContext = createServer(expressApp);
+
+        this.socketServerInit(this.serverContext);
+
+        this.serverContext.listen(this.port, async () => {
+            // tslint:disable-next-line:no-console
+            console.log(`started server at http://localhost:${this.port}`);
+        });
+    }
+
+    /**
+     * Not private since tests will be required to confiugre their own instance of a @type {serverApplication}
+     * @param httpServer HTTP server where socket server will be attached
+     * @returns expressJs context
+     */
+    socketServerInit(httpServer: HTTPServer): SocketServer {
+        const FRONTEND_URL = process.env.FRONTEND_URL || "http://localhost:3000";
+
+        const io = new SocketServer<ClientToServerEvents, ServerToClientEvents>(httpServer, {
+            cors: {
+                origin: FRONTEND_URL
+            }
+        });
+        
+        // Setup and configure socket routes and events
+
+        io.on("connection", (socket) => {
+
+            // tslint:disable-next-line:no-console
+            console.log("Connection Established");
```

Comment:
I'm curious now. Would most cloud providers keep logs for us? Or is this only really useful for local debugging?

Reply:
Currently only useful for local debugging. Wanted to see if everything worked including CORS. I'm sure Heroku will provide logs for us though.